server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/SearchSnippetGetMethod.scala

```diff
@@ -0,0 +1,108 @@
+      .map(invocationResult => InvocationWithContext(invocationResult, invocation.processingContext))
+
+  private def processRequest(mailboxSession: MailboxSession,
+                             invocation: Invocation,
+                             request: SearchSnippetGetRequest,
+                             capabilities: Set[CapabilityIdentifier]): SMono[Invocation] =
+    getSearchSnippet(request, mailboxSession, capabilities)
+      .map(searchSnippetList => SearchSnippetGetResponse.from(request.accountId, searchSnippetList.toList, request.emailIds))
+      .map(response => Invocation(
+        methodName = methodName,
+        arguments = Arguments(serializer.serialize(response).as[JsObject]),
+        methodCallId = invocation.methodCallId))
+
+  private def getSearchSnippet(request: SearchSnippetGetRequest,
+                               mailboxSession: MailboxSession,
+                               capabilities: Set[CapabilityIdentifier]): SMono[Seq[SearchSnippet]] = {
+    val validatedMessageId: List[MessageId] = request.emailIds.flatMap(asMessageId(_).toOption)
+    if (request.emptyRequest || validatedMessageId.isEmpty) {
+      SMono.just(Seq.empty)
+    } else {
+      buildMultiMailboxesSearchQuery(request, validatedMessageId, capabilities, mailboxSession)
+          .fold(e => SMono.error(e),
+            multiMailboxesSearchQuery => SFlux(searchHighlighter.highlightSearch(validatedMessageId.asJava, multiMailboxesSearchQuery, mailboxSession))
+              .collectSeq())
+        .switchIfEmpty(SMono.just(Seq.empty))
+    }
+  }
+
+  private def asMessageId(id: UnparsedEmailId): Either[(UnparsedEmailId, IllegalArgumentException), MessageId] =
+    Try(messageIdFactory.fromString(id.id.value))
+      .toEither
+      .left.map(error => (id, new IllegalArgumentException(s"Can not parse UnparsedEmailId(${id.id.value}) as MessageId", error)))
+
+  private def buildMultiMailboxesSearchQuery(request: SearchSnippetGetRequest,
+                                             messageIds: List[MessageId],
+                                             capabilities: Set[CapabilityIdentifier],
+                                             session: MailboxSession): Either[UnsupportedOperationException, MultimailboxesSearchQuery] = {
+    val messageIdsCriterion = new ConjunctionCriterion(SearchQuery.Conjunction.OR, messageIds.map(SearchQuery.hasMessageId).asJava)
+
+    request.tryFilterAsSearchQuery
+      .map(searchQueryBuilder => searchQueryBuilder.andCriteria(messageIdsCriterion).build())
+      .map(searchQuery => MultimailboxesSearchQuery.from(searchQuery)
+        .inNamespace(queryNamespace(capabilities, session))
+        .build())
+  }
+
+  private def queryNamespace(capabilities: Set[CapabilityIdentifier], session: MailboxSession): Namespace =
+    if (capabilities(CapabilityIdentifier.JAMES_SHARES)) new AccessibleNamespace()
+    else new PersonalNamespace(session)
```

Comment:
This behavioural change needs to be documented in the JMAP sharing extension.

(Adding JAMES_SHARES capability allows one to render search snippets not in his account)